sweep/sweeper.go

```diff
@@ -20,51 +22,53 @@ type Sweeper struct {
 }
 
 // CreateUnsignedTaprootKeySpendSweepTx creates a taproot htlc sweep tx using
-// keyspend. Returns the raw unsigned txn and the sighash or an error.
+// keyspend. Returns the raw unsigned txn, the psbt serialized txn, the sighash
+// or an error.
 func (s *Sweeper) CreateUnsignedTaprootKeySpendSweepTx(
 	ctx context.Context, lockTime uint32,
 	htlc *swap.Htlc, htlcOutpoint wire.OutPoint,
 	amount, fee btcutil.Amount, destAddr btcutil.Address) (
-	*wire.MsgTx, []byte, error) {
+	*wire.MsgTx, []byte, []byte, error) {
 
 	if htlc.Version != swap.HtlcV3 {
-		return nil, nil, fmt.Errorf("invalid htlc version")
+		return nil, nil, nil, fmt.Errorf("invalid htlc version")
 	}
 
-	// Compose tx.
-	sweepTx := wire.NewMsgTx(2)
-	sweepTx.LockTime = lockTime
-
-	// Add HTLC input.
-	sweepTx.AddTxIn(&wire.TxIn{
-		PreviousOutPoint: htlcOutpoint,
-		SignatureScript:  htlc.SigScript,
-	})
-
 	// Add output for the destination address.
 	sweepPkScript, err := txscript.PayToAddrScript(destAddr)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 
-	sweepTx.AddTxOut(&wire.TxOut{
-		PkScript: sweepPkScript,
-		Value:    int64(amount - fee),
-	})
+	// Compose tx.
+	sweepTx := &wire.MsgTx{
+		Version: 2,
+		TxIn: []*wire.TxIn{{
+			PreviousOutPoint: htlcOutpoint,
+			SignatureScript:  htlc.SigScript,
+		}},
+		TxOut: []*wire.TxOut{{
+			Value:    int64(amount - fee),
+			PkScript: sweepPkScript,
+		}},
+		LockTime: lockTime,
+	}
 
-	// We need our previous outputs for taproot spends, and there's no
-	// harm including them for segwit v0, so we always include our prevOut.
-	prevOut := []*wire.TxOut{
-		{
-			Value:    int64(amount),
-			PkScript: htlc.PkScript,
-		},
+	packet, err := psbt.NewFromUnsignedTx(sweepTx)
```

Comment:
Should we set the `WitnessUtxo` field on the input as well?

Reply:
done